### PR TITLE
doc: Fix some misspellings

### PR DESCRIPTION
**Pull request by RandyMcMillan**
Here is a more thorough lint-spelling update.
This PR takes care of easy to fix spelling errors to clean up the linting stages.
There are misspellings coded into the functional tests.
That is a whole separate job within itself.

### CODE REVIEW

**Review by hebasto — COMMENTED** [review]
Concept ACK ac6d97fc4a1cd04db20fd773a05cd1eb0eebc373, tested on Linux Mint 19.2:

```
$ pip3 show codespell | grep Version
Version: 1.16.0

$ ./test/lint/lint-spelling.sh 
src/test/base32_tests.cpp:14: fo  ==> of, for
src/test/base64_tests.cpp:14: fo  ==> of, for
test/lint/lint-locale-dependence.sh:100: stoll  ==> still
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/lint-spelling.ignore-words.txt
```

EDITED: @RandyMcMillan FWIW: `Status: Draft` means PR is not ready to be reviewed.

EDITED 2: IMHO, the commit message could be more expressive like `test: fix all typos` ;)

**Comment by laanwj** [comment]
Concept ACK, but from now on, stop opening multiple PRs for everything. If you need to figure out how github works, please experiment with a repository of your own instead of our production repo.

**Comment by hebasto** [comment]
These words could be dropped from ignore list:
https://github.com/bitcoin/bitcoin/blob/9641366950276da88af626d0898676195df8d83a/test/lint/lint-spelling.ignore-words.txt#L1

https://github.com/bitcoin/bitcoin/blob/9641366950276da88af626d0898676195df8d83a/test/lint/lint-spelling.ignore-words.txt#L6

---
`mut` could be dropped with renaming `s/mut/m_mutex/` in `threadinterrupt.{h|cpp}`.

---

`useable`, `unparseable`, `cachable` could be dropped with proper fixes in the code base. However, I'm not sure about other devs opinion ;)

---

`homogenous` could be dropped with a proper fix in `tinyformat.h`. Also please see https://github.com/bitcoin/bitcoin/pull/12716#discussion_r176130024.

File: test/lint/lint-spelling.sh [hunk 1]
```diff
@@ -15,6 +15,6 @@ if ! command -v codespell > /dev/null; then
 fi
 
 IGNORE_WORDS_FILE=test/lint/lint-spelling.ignore-words.txt
-if ! codespell --check-filenames --disable-colors --quiet-level=7 --ignore-words=${IGNORE_WORDS_FILE} $(git ls-files -- ":(exclude)build-aux/m4/" ":(exclude)contrib/seeds/*.txt" ":(exclude)depends/" ":(exclude)doc/release-notes/" ":(exclude)src/leveldb/" ":(exclude)src/qt/locale/" ":(exclude)src/secp256k1/" ":(exclude)src/univalue/"); then
+if ! codespell --check-filenames --disable-colors --quiet-level=7 --ignore-words=${IGNORE_WORDS_FILE} $(git ls-files -- ":(exclude)build-aux/m4/" ":(exclude)contrib/seeds/*.txt" ":(exclude)depends/" ":(exclude)doc/release-notes/" ":(exclude)src/leveldb/" ":(exclude)src/qt/locale/" ":(exclude)src/qt/bitcoin_locale.qrc" ":(exclude)src/secp256k1/" ":(exclude)src/univalue/"); then
```
Comment (practicalswift) [inline test/lint/lint-spelling.sh]: Nit: Could replace `":(exclude)src/qt/bitcoin_locale.qrc"` with `":(exclude)*.qrc"`

**Comment by practicalswift** [comment]
ACK ac6d97fc4a1cd04db20fd773a05cd1eb0eebc373 modulo addressing of the feedback from @hebasto :)

**Comment by hebasto** [comment]
Here are some recent similar PRs:
- #16186
- #16767

**Comment by RandyMcMillan** [comment]
@laanwj - I apologize for the PR noise. For some reason I can't reopen PRs. It is probably something I did. (forced push setting in your admin settings) - I have since figured out my workflow. So if it is an admin setting on your side - I apologize and have since implemented my own build environment and will only create PRs once I have tested my changes personally FIRST! Again sincere apology.  ditto to @MarcoFalke and the rest of the members - @practicalswift @dongcarl @promag etc...

File: test/lint/lint-spelling.sh (hunk 1 above)
Comment (RandyMcMillan) [reply test/lint/lint-spelling.sh]: Thanks @practicalswift - awesome suggestion.

Comment (RandyMcMillan) [reply test/lint/lint-spelling.sh]: Included this suggestion in the commit. thanks again.

**Comment by RandyMcMillan** [comment]
> These words could be dropped from ignore list:
> 
> https://github.com/bitcoin/bitcoin/blob/9641366950276da88af626d0898676195df8d83a/test/lint/lint-spelling.ignore-words.txt#L1
> 
> https://github.com/bitcoin/bitcoin/blob/9641366950276da88af626d0898676195df8d83a/test/lint/lint-spelling.ignore-words.txt#L6
> 
> `mut` could be dropped with renaming `s/mut/m_mutex/` in `threadinterrupt.{h|cpp}`.

I left the words you were unsure about out until we have more feed back.  

> `useable`, `unparseable`, `cachable` could be dropped with proper fixes in the code base. However, I'm not sure about other devs opinion ;)
> 
> `homogenous` could be dropped with a proper fix in `tinyformat.h`. Also please see [#12716 (comment)](https://github.com/bitcoin/bitcoin/pull/12716#discussion_r176130024).

**Comment by practicalswift** [comment]
ACK d17d888551d928642fcf76cbf64e7aa278426143

**Comment by hebasto** [comment]
Tested d17d888551d928642fcf76cbf64e7aa278426143:

```
$ ./test/lint/lint-spelling.sh | grep mut
src/threadinterrupt.cpp:25: mut  ==> must, mutt, moot
src/threadinterrupt.cpp:33: mut  ==> must, mutt, moot
src/threadinterrupt.h:32: mut  ==> must, mutt, moot
```

@RandyMcMillan Did you forget to rename `mut` to `m_mutex` as [suggested](https://github.com/bitcoin/bitcoin/pull/17351#issuecomment-549032608)?

---

> I also removed /bitcoin/contrib/guix/manifest.scm from the commit.

Why? The goal of the mentioned commit is orthogonal to the goal of this PR.
I'd also prefer a complete solution over partial one, if it is feasible.

---

There is a misspelling in the commit message "doc:Fix some mispellings" ;)

**Comment by RandyMcMillan** [comment]
> ```
> $ ./test/lint/lint-spelling.sh | grep mut

@hebasto - speaking of grep - we need to add an alias to ggrep in the Mac environment. Brew doesnt do it automatically - this will take care on a linting issue in the Mac build.

**Comment by promag** [comment]
Concept ACK, no longer draft?

CI failing with, for instance:
```
Running test/util/bitcoin-util-test.py...
/usr/bin/python3.6 ../test/util/bitcoin-util-test.py
2019-11-03 20:06:22,888 - ERROR - Error mismatch:
Expected: error: Uncompressed pubkeys are not usable for SegWit outputs
Received: error: Uncompressed pubkeys are not useable for SegWit outputs
```

**Comment by RandyMcMillan** [comment]
NOTE: There are misspellings built into the functional tests. That is a whole job with in itself.

Preliminary build pass:
https://travis-ci.org/RandyMcMillan/bitcoin/builds/606892791

@hebasto @promag Yes, I believe it is ready. 

I appreciate your patience and feedback. I know this stuff is tedious and probably not a priority.
But I am getting familiar with the code base this way and somebody has to do the broom pushing - so thanks!

**Comment by hebasto** [comment]
@RandyMcMillan Please do not touch files in `doc/man/` as they are generated by a script, and historical release notes in `doc/release-notes/`.

**Review by practicalswift — APPROVED** [review]
ACK ac831339cbfa65b1f7576c53b5d9a94841db9868 -- diff looks correct

**Comment by RandyMcMillan** [comment]
@hebasto - done. But can I make the point - by NOT fixing "historical" spelling issues in the release notes - we are also choosing to allow them and ***NEW ERRORS*** to persist into the future? If we fix the old issues ***NOW*** we can enforce better policy and linting moving forward? Couldn't we even implement a build step that focuses solely on these types of issues or remove any rules that ignore documentation? (that was the thought behind editing the release notes to begin with)

P.S. Can you point me in the direction of the scripts that generate the man pages?

Thanks! 👍

**Comment by laanwj** [comment]
> But can I make the point - by NOT fixing "historical" spelling issues in the release notes - we are also choosing to allow them and NEW ERRORS to persist into the future? 

They are historical documents, which have been published to many places besides this repository. Fixing errors there would be like going to a museum and scribbling in a 12th century bible, for example :smile: It's no reason to allow new errors in new files, just, what has been done has been done.

**Pull request merged**